tap_shopify/streams/compatibility/product_compatibility.py

```diff
@@ -0,0 +1,145 @@
+import json
+import os
+import shopify.resources
+
+class ProductCompatibility():
+    def __init__(self, graphql_product):
+        """Initialize with a GraphQL product object."""
+        self.graphql_product = graphql_product
+        self.admin_graphql_api_id = graphql_product["id"]
+        self.product_id = self._extract_int_id(graphql_product["id"])
+
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+        value_map_path = os.path.join(current_dir, "value_maps", "product.json")
+        with open(value_map_path, 'r') as file:
+            self.value_map = json.load(file)
+
+    def metafields(self, _options=None, **kwargs):
```

Comment:
This was added to support the metafields stream, which calls this method [here](https://github.com/hotgluexyz/tap-shopify/blob/feature/hgi-6654/tap_shopify/streams/metafields.py#L22).

Unfortunately, it seems that the request `shopify.resources.Metafield.find` issues looks like this:

```
https://hotglue-testing-1.myshopify.com/admin/api/2024-01/products/7903137038556/metafields.json?limit=175&since_id=1
```

Which calls the `/products` endpoint. Looking into how to resolve this.